Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs

```diff
@@ -711,3 +712,15 @@ updateAlternateMobileNumberAndCode person = do
         PersonUpdatedAt =. val now
       ]
     where_ $ tbl ^. PersonTId ==. val (toKey person.id)
+
+updateMediaId :: Id Person -> Maybe (Id MediaFile) -> SqlDB ()
```

Comment:
why media file is of maybe type, do you wanted to remove the id value if mediafile id is empty ?

Otherwise you can do this.

`updateMediaId driverId mediaFileId =
  Esq.update $ \tbl ->
when
    (isJust mediaFileId)
 do
    set
      tbl
      [PersonMediaId =. val mediaIdValue]
    where_ $ tbl ^. PersonTId ==. val (toKey $ cast driverId)
  where
    mediaIdValue = case mediaFileId of
      Just mediaId -> Just $ toKey mediaId
      Nothing -> Nothing`

it will override only when you pass a media id.

Reply:
Just add `when (isJust mediaFileId)` while updating

Reply:
mediaFileId is chosen of type maybe so as to nullify the foreign key in Person table by passing Nothing (in place of mediaFileId), before deleting the entry from media table.